test/test_plugins.py

```diff
@@ -0,0 +1,79 @@
+# coding: utf-8
+import importlib
+import sys
+import unittest
+from importlib import invalidate_caches
+from pathlib import Path
+import os
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+TEST_DATA_DIR = Path(os.path.dirname(os.path.abspath(__file__)), 'testdata')
+sys.path.append(str(TEST_DATA_DIR))
+invalidate_caches()
+
+from yt_dlp.plugins import PACKAGE_NAME, directories, load_plugins
+
+
+class TestPlugins(unittest.TestCase):
+
+    TEST_PLUGIN_DIR = TEST_DATA_DIR / f'{PACKAGE_NAME}'
+
+    def test_plugin_directory_structure(self):
+        # XXX: doesn't test much?
+        self.assertFalse(self.TEST_PLUGIN_DIR.joinpath('__init__.py').exists())
+        self.assertTrue(self.TEST_PLUGIN_DIR.joinpath('extractor').is_dir())
+        self.assertFalse(self.TEST_PLUGIN_DIR.joinpath('extractor', '__init__.py').exists())
+        self.assertTrue(self.TEST_PLUGIN_DIR.joinpath('postprocessor').is_dir())
+        self.assertFalse(self.TEST_PLUGIN_DIR.joinpath('postprocessor', '__init__.py').exists())
```

Comment:
this is testing the testdata itself? Can be removed imo

Reply:
pretty much. I don't see the point of it